### PR TITLE
docs(typehead): add demo for typeaheadOnSelect output #3681

### LINKED ISSUE #3681: docs(typehead): add demo for typeaheadOnSelect output
Add demo for typeaheadOnSelect output (related to #3342)

### PR DESCRIPTION
**Pull request by IraErshova**
Add demo for typeaheadOnSelect output. To reproduce demo, please select one of the options and then you can see the selected option above the input ("Selected option"). [See screenshot ](http://joxi.ru/brRggRgHQZ9nam)

Close #3681

# PR Checklist
Before creating new PR, please take a look at checklist below to make sure that you've done everything that needs to be done before we can merge it.

 - [ ] read and followed the [CONTRIBUTING.md](https://github.com/valor-software/ngx-bootstrap/blob/development/CONTRIBUTING.md) guide.
 - [ ] built and tested the changes locally.
 - [ ] added/updated API documentation.
 - [ ] added/updated demos.

### CODE REVIEW

File: demo/src/app/components/+typeahead/typeahead-section.list.ts [hunk 1]
```diff
@@ -83,6 +84,16 @@ export const demoComponentContent: ContentSection[] = [
         component: require('!!raw-loader?lang=typescript!./demos/scrollable/scrollable.ts'),
         html: require('!!raw-loader?lang=markup!./demos/scrollable/scrollable.html'),
         outlet: DemoTypeaheadScrollableComponent
+      },
+      {
+        title: 'On select',
+        anchor: 'on-select',
+        description: `
+          <p>Fired when option was selected, return object with data of this option</p>
```
Comment (IlyaSurmay) [inline demo/src/app/components/+typeahead/typeahead-section.list.ts]: Fired when an option was selected, returns an object with this option.

**Review by IlyaSurmay — CHANGES_REQUESTED** [review]

File: demo/src/app/components/+typeahead/typeahead-section.list.ts (hunk 1 above)
Comment (IlyaSurmay) [reply demo/src/app/components/+typeahead/typeahead-section.list.ts]: please change this in the jsdoc as well

**Review by IlyaSurmay — DISMISSED** [review]

**Comment by YevheniiaMazur** [comment]
Please, provide more detailed description, how to test this pr (when this event calls, if it is event, etc)

**Comment by YevheniiaMazur** [comment]
Tested, looks good

File: demo/src/app/components/+typeahead/demos/on-select/on-select.ts [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+    { id: 45, name: 'Utah', region: 'West' },
+    { id: 46, name: 'Vermont', region: 'Northeast' },
+    { id: 47, name: 'Virginia', region: 'South' },
+    { id: 48, name: 'Washington', region: 'South' },
+    { id: 49, name: 'West Virginia', region: 'South' },
+    { id: 50, name: 'Wisconsin', region: 'Midwest' },
+    { id: 51, name: 'Wyoming', region: 'West' }
+  ];
+
+  onSelect(event: any): void {
```
Comment (valorkin) [inline demo/src/app/components/+typeahead/demos/on-select/on-select.ts]: @IraErshova event type

**Review by valorkin — CHANGES_REQUESTED** [review]

**Pull request merged**